frappe/public/js/frappe/widgets/widget_dialog.js

```diff
@@ -397,7 +397,11 @@ class ShortcutDialog extends WidgetDialog {
 							const views = ["List", "Report Builder", "Dashboard", "New"];
 							if (meta.is_tree === "Tree") views.push("Tree");
 							if (frappe.boot.calendars.includes(doctype)) views.push("Calendar");
-
+							frappe.db.get_value("Kanban Board", { reference_doctype: doctype }, "name", (response) => {
+								if (response && response.name) {
+								  views.push("Kanban");
+								}
+							  });
```

Comment:
This will not work, because df_property will be set before the get_value promise is resolved.
I will suggest to use async await here

Also, hide the doc_view until and unless doctype(link_to) is not set